### PR TITLE
ref(joins): Joined storages do not have a storage key

### PR DESCRIPTION
**Pull request by lynnagara**
Joined storages should not have a storage key (and possibly shouldn't be
considered a storage at all).

### CODE REVIEW

File: snuba/datasets/storage.py [hunk 1]
```diff
@@ -31,16 +31,10 @@ class Storage(ABC):
...
-        self, storage_key: StorageKey, storage_set_key: StorageSetKey, schema: Schema
-    ):
-        self.__storage_key = storage_key
+    def __init__(self, storage_set_key: StorageSetKey, schema: Schema):
         self.__storage_set_key = storage_set_key
         self.__schema = schema
 
-    def get_storage_key(self) -> StorageKey:
-        return self.__storage_key
-
```
Comment (fpacifici) [inline snuba/datasets/storage.py:43]: Why this change? It does not seem related to joins

Comment (lynnagara) [reply snuba/datasets/storage.py:43]: I moved it to line 112 since only table storages have a storage key now. I could leave it here and return None for joined storages.

**Review by fpacifici — APPROVED** [review]
